crates/cast/src/lib.rs

```diff
@@ -981,6 +982,119 @@ impl<P: Provider<AnyNetwork>> Cast<P> {
         Ok(res)
     }
 
+    pub async fn filter_logs_chunked(&self, filter: Filter, chunk_size: u64) -> Result<String> {
+        let logs = self.get_logs_chunked(&filter, chunk_size).await?;
+
+        let res = if shell::is_json() {
+            serde_json::to_string(&logs)?
+        } else {
+            let mut s = vec![];
+            for log in logs {
+                let pretty = log
+                    .pretty()
+                    .replacen('\n', "- ", 1) // Remove empty first line
+                    .replace('\n', "\n  "); // Indent
+                s.push(pretty);
+            }
+            s.join("\n")
+        };
+        Ok(res)
+    }
+
+    async fn get_logs_chunked(&self, filter: &Filter, chunk_size: u64) -> Result<Vec<Log>> {
+        // Try the full range first
+        match self.provider.get_logs(filter).await {
+            Ok(logs) => Ok(logs),
+            Err(e) => {
```

Comment:
this section is a bit hard to follow, could we rewrite this a bit so that it's more obvious what it does